spec/requests/v1/pacbio/pools_spec.rb

```diff
@@ -360,5 +363,45 @@
       end
     end
 
+    context 'when there is an associated run' do
+      let!(:pool) { create(:pacbio_pool) }
+      let!(:updated_library) { pool.libraries.first }
+      let!(:plate) { create(:pacbio_plate) }
+      let!(:well) { create(:pacbio_well, pools: [pool], plate: plate) }
+      let(:body) do
+        {
+          data: {
+            type: 'pools',
+            id: pool.id.to_s,
+            attributes: {
+              library_attributes: [
+                {
+                  id: updated_library.id.to_s,
+                  pacbio_request_id: updated_library.pacbio_request_id.to_s,
+                  template_prep_kit_box_barcode: 'LK12345',
+                  tag_id: tag.id,
+                  volume: 1,
+                  concentration: 1,
+                  insert_size: 100
+                }
+              ],
+              volume: '200',
+              concentration: '22',
+              template_prep_kit_box_barcode: '100',
+              insert_size: '11',
+              created_at: '2021-08-04T14:35:47.208Z',
+              updated_at: '2021-08-04T14:35:47.208Z'
+            }
+          }
+        }.to_json
+      end
+      
+      it 'publishes a message' do
+        expect(Messages).to receive(:publish).with(pool.sequencing_plates, having_attributes(pipeline: 'pacbio'))
+        patch v1_pacbio_pool_path(pool), params: body, headers: json_api_headers
+        expect(response).to have_http_status(:success), response.body
```

Comment:
Does it still publish a message if a patch has been performed but not data has actually changed? I guess that would do an update in MLWH, but shouldn't be harmful

Reply:
Yes. It has no side effect so think it is safe. I suppose we could use the dirty active record method but I thought it was adding complexity.